tests/unit/cocoon/test_cocoon_complex_instrument.py

```diff
@@ -0,0 +1,204 @@
+                # coupon payment currency
+                currency="USD",
+                # semi-annual coupon payments
+                payment_frequency="6M",
+                # using an Actual/365 day count convention (other options : Act360, ActAct, ...
+                day_count_convention="Thirty360",
+                # modified following rolling convention (other options : ModifiedPrevious, NoAdjustment, EndOfMonth,...)
+                roll_convention="MF",
+                # no holiday calendar supplied
+                payment_calendars=[],
+                reset_calendars=[],
+                settle_days=2,
+                reset_days=2,
+            ),
+            identifiers={},
+            instrument_type="Bond",
+        )
+
+        instrument_request = {
+            "ClientInternal: imd_05017030132": models.InstrumentDefinition(
+                # instrument display name
+                name="DISNEY_7_2032",
+                # unique instrument identifier
+                identifiers={
+                    "ClientInternal": models.InstrumentIdValue("imd_05017030132"),
+                    "Isin": models.InstrumentIdValue("US25468PBW59"),
+                },
+                # instrument definition
+                definition=instrument_definition,
+            )
+        }
+
+        return self.instruments_api.upsert_instruments(instrument_request)
+
+    @parameterized.expand(
+        [
+            [
+                "A standard successful load of a complex instrument using a full bond definition",
+                "TestScope1",
+                "data/global-fund-single-bond-master.csv",
+                {
+                    "name": "Name",
+                    "instrument_type": "Instrument Type",
+                    "start_date": "Issue Date",
+                    "maturity_date": "Maturity Date",
+                    "dom_ccy": "Currency",
+                    "flow_conventions": {
+                        "currency": "Currency",
+                        "payment_frequency": "Payment Frequency",
+                        "day_count_convention": "Day Count Convention",
+                        "roll_convention": "Roll Convention",
+                        "payment_calendars": "Payment Calendars",
+                        "reset_calendars": "Reset Calendars",
+                        "reset_days": "Reset Days",
+                    },
+                    "principal": "Principal",
+                    "coupon_rate": "Coupon",
+                },
+                {},
+                {"Isin": "ISIN", "ClientInternal": "Client Internal"},
+                [],
+                "TestPropertiesScope1",
+            ],
+            [
+                "A standard successful load of a complex instrument using a full bond definition",
+                "TestScope1",
+                "data/global-fund-fixed-income-master.csv",
+                {
+                    "name": "Name",
```

Comment:
I don't see where you would ever have a dictionary which has `instrument_type`  as a key based on this mapping here?